Algorithms/GEMETD.py

```diff
@@ -0,0 +1,56 @@
+from Algorithms.ETDLB import ETDLB
+from Algorithms.BaseTD import BaseTD
+import numpy as np
+
+
+class GEMETD(ETDLB):
+    """
+    An ETD(0) implementation that uses GEM (aka GTD2(0) with x and x_p switched) to estimate emphasis.
+    """
+    def __init__(self, task, **kwargs):
+        super().__init__(task, **kwargs)
+        self.beta = self.task.GAMMA
+        self.gem_alpha = self.get('gem_alpha')  # Step size for GEM weights.
+        self.gem_eta = self.get('gem_eta')  # Regularization parameter for GEM; not needed for a fixed target policy.
+        self.k = np.zeros(self.task.num_features)  # Auxiliary weights for GTD2.
+        self.u = np.zeros(self.task.num_features)  # Main weights for GTD2.
+        if self.task.num_policies > 1:
+            self.k = np.zeros((self.task.num_policies, self.task.num_features))
+            self.u = np.zeros((self.task.num_policies, self.task.num_features))
+
+    @staticmethod
+    def related_parameters():
+        return ['alpha', 'gem_alpha', 'gem_eta']
```

Comment:
I think I get it. We need two step sizes for two weight vectors. Is that right? In that case, we might want to use ``self.alpha`` for the main weight vector, and ``self.eta*self.alpha`` for the secondary weight vector. What do you think?

Reply:
We can use the ``alpha_v = self.compute_second_step_size()`` function to get the second step size.

Reply:
``alpha_v=self.compute_second_step_size()`` should simply return the second step-size parameter. The first step-size parameter should already be in ``self.alpha``

Reply:
There's actually three weight vectors in GEM-ETD: `self.u` to estimate the emphasis for a given feature vector, `self.k` to stabilize the updates to `self.u` (similar to how `self.v` stabilizes the updates to `self.w` in GTD2(0), but for emphasis instead of value), and `self.w` to estimate the value function. In theory there could be 3 different step sizes, but the GEM paper uses the same step size for both GEM weight vectors. So we need `alpha` for `self.w` and `gem_alpha` for `self.u` and `self.k`.

Reply:
This is pretty clear now. Thanks.